editor/components/post-featured-image/index.js

```diff
@@ -77,13 +84,10 @@ const applyConnect = connect(
 	}
 );
 
-const applyWithAPIData = withAPIData( ( { featuredImageId } ) => {
-	if ( ! featuredImageId ) {
-		return {};
-	}
-
+const applyWithAPIData = withAPIData( ( { featuredImageId, postTypeName } ) => {
 	return {
-		media: `/wp/v2/media/${ featuredImageId }`,
+		media: featuredImageId ? `/wp/v2/media/${ featuredImageId }` : undefined,
+		postType: postTypeName ? `/wp/v2/types/${ postTypeName }?context=edit` : undefined,
```

Comment:
Unfortunately you cannot rely on the post type slug, since a plugin author can define a different `rest_base` for their custom post types:

https://developer.wordpress.org/rest-api/extending-the-rest-api/adding-rest-api-support-for-custom-content-types/

You can see this in core behavior, where `/wp/v2/types/attachment` is incorrect, since the `rest_base` for the `attachment` post type has been assigned as `media`.

This is why the `type` helper from `withAPIData` second argument should be used.

Reply:
Hi @aduth, thank you for having a look into this. I used the post type slug because it looks like this specific endpoint relies on it instead of the type.
E.g .../wp-json/wp/v2/types/attachment is a valid request, while .../wp-json/wp/v2/types/media is not a valid request.
The some for posts the type is 'posts' and if we use it we create an invalid request, while we use the slug the request is valid.

Reply:
Aha! That makes sense, sorry for the quick instinctual reaction 😄

Reply:
No problem at all, at the first I also tried to use post type and then I found it really really strange when it was not working as I expected :D